File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hm = {}

        for i in range(len(nums)):
            diff = target - nums[i]

            if diff in hm:
                return i, hm[diff]
            
            else:
                hm[nums[i]] = i
        
        

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> hm = new HashMap<>();
        int diff;
        
        for(int i=0; i<nums.length; i++){
            diff = target - nums[i];
            
            if (hm.containsKey(diff)){
                return new int[]{i, hm.get(diff)};
            }
            
            hm.put(nums[i], i);
        }
        
        return null;
    }
}